player/command.c

```diff
@@ -515,24 +515,27 @@ static int mp_property_media_title(void *ctx, struct m_property *prop,
                                    int action, void *arg)
 {
     MPContext *mpctx = ctx;
-    char *name = NULL;
+    const char* name = NULL;
     if (mpctx->opts->media_title)
         name = mpctx->opts->media_title;
+    if ((!name || !name[0]) && mpctx->demuxer) {
+        name = mp_tags_get_str(mpctx->demuxer->metadata, "service_name");
+        if (!name || !name[0]){
+            name = mp_tags_get_str(mpctx->demuxer->metadata, "title");
+            if (!name || !name[0])
+                name = mp_tags_get_str(mpctx->demuxer->metadata, "icy-title");
+        }
+    }
+    struct playlist_entry *const pe = mpctx->playing;
+    if (pe) {
+        if (!name || !name[0]){
+            name = pe->title;
+        } else if (!pe->title) {
+            pe->title = talloc_strdup(pe, name);
```

Comment:
This is the property handler for `media-title`.
1) why does this _modify_ a playlist entry?
2) how does this work if nobody touches `media-title`?

Reply:
If we want to set the playlist title to the media title it needs to be done in the `media-title` property handler not the playlist one since it would need to update itself anytime the `media-title` changes.

Reply:
You're right, I thought it was fine because the handler usually gets a ton of calls, but handler never gets called when running `mpv --no-config --no-terminal --no-osc --title=foo <some file>`.

Any suggestions on a better way of implementing that?

Reply:
Not sure, maybe in `update_demuxer_properties` or some other place that deals with playlist changes.